great_expectations/datasource/fluent/interfaces.py

```diff
@@ -472,6 +476,10 @@ def add_asset(self, asset: _DataAssetT) -> _DataAssetT:
         # See the comment in DataAsset for more information.
         asset._datasource = self
 
+        if not connect_options:
+            connect_options = {}
+        self._build_data_connector(asset, **connect_options)
```

Comment:
@Kilo59 I am not seeing how it works whereby asset is of type `_DataAssetT`, which is bound to the `DataAsset` type, whereas the `DataConnector` constructor accepts the `data_asset_name` (whose type is a string).  Please let me know where/how this is reconciled.  Thank you.

Reply:
See the concrete implementation of `_build_data_connector()` `pandas_filesystem_datasource.py`

Reply:
@Kilo59 Got it -- thanks; looks great (did not see before among all the red of the deleted code).